### PR TITLE
cosmosdb metrics terminate early

### PR DESCRIPTION
**Pull request by mjudeikis**
Fixes: https://github.com/Azure/ARO-RP/issues/126

### CODE REVIEW

File: pkg/metrics/statsd/cosmosdb/metrics.go [hunk 1]
```diff
@@ -34,6 +34,9 @@ func (t *tracerRoundTripper) RoundTrip(req *http.Request) (resp *http.Response,
 	start := time.Now()
 
 	defer func() {
+		if resp == nil {
```
Comment (jim-minter) [inline pkg/metrics/statsd/cosmosdb/metrics.go]: you are not ever setting resp! ;-)

Comment (mjudeikis) [reply pkg/metrics/statsd/cosmosdb/metrics.go]: `func (t *tracerRoundTripper) RoundTrip(req *http.Request) (resp *http.Response, err error) ` naked return ?

Comment (jim-minter) [reply pkg/metrics/statsd/cosmosdb/metrics.go]: Good point!  hehe.  This makes me think that we should add another metric that counts errors, wdyt?

Comment (mjudeikis) [reply pkg/metrics/statsd/cosmosdb/metrics.go]: Yes, this was on my head too but I wanted to remove panics first

Comment (mjudeikis) [reply pkg/metrics/statsd/cosmosdb/metrics.go]: https://github.com/Azure/ARO-RP/issues/131

**Pull request merged**